projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/strategy/datajob/JobFieldStrategyByLastExecutionStatus.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.graphql.strategy.datajob;
+
+import com.vmware.taurus.controlplane.model.data.DataJobExecution;
+import com.vmware.taurus.service.graphql.model.Criteria;
+import com.vmware.taurus.service.graphql.model.Filter;
+import com.vmware.taurus.service.graphql.model.V2DataJob;
+import com.vmware.taurus.service.graphql.strategy.FieldStrategy;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Comparator;
+import java.util.function.Predicate;
+
+@Component
+public class JobFieldStrategyByLastExecutionStatus extends FieldStrategy<V2DataJob> {
+
+   private static final Comparator<V2DataJob> COMPARATOR_DEFAULT = Comparator.comparing(
+           JobFieldStrategyByLastExecutionStatus::getLastExecutionStatus);
+
+   @Override
+   public JobFieldStrategyBy getStrategyName() {
+      return JobFieldStrategyBy.DEPLOYMENT_LAST_EXECUTION_STATUS;
+   }
+
+   @Override
+   public Criteria<V2DataJob> computeFilterCriteria(@NonNull Criteria<V2DataJob> criteria, @NonNull Filter filter) {
+      Predicate<V2DataJob> predicate = criteria.getPredicate();
+
+      if (filterProvided(filter)) {
+         predicate = predicate.and(computeSearchCriteria(filter.getPattern()));
+      }
+
+      return new Criteria<>(predicate, detectSortingComparator(filter, COMPARATOR_DEFAULT, criteria));
+   }
+
+   @Override
+   public Predicate<V2DataJob> computeSearchCriteria(@NonNull String searchStr) {
+      try {
+         DataJobExecution.StatusEnum executionStatus = DataJobExecution.StatusEnum.fromValue(searchStr);
+         return dataJob -> executionStatus.equals(getLastExecutionStatus(dataJob));
+      } catch (IllegalArgumentException e) {
+         // searching with non-compatible for the strategy string, ignoring search
+         return x -> false;
+      }
+   }
+
+   private static DataJobExecution.StatusEnum getLastExecutionStatus(V2DataJob dataJob) {
+      if (CollectionUtils.isEmpty(dataJob.getDeployments())) {
+         return null;
+      }
+      // TODO support for multiple deployments
+      return dataJob.getDeployments().stream().findFirst().orElseThrow().getLastExecutionStatus();
```

Comment:
In already implemented code we perform a null check for deployments e.g:

versatile-data-kit/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java

Lines 81 to 83 in f086df8
 allDataJob.forEach(dataJob -> { 
    if (dataJob.getDeployments() != null) { 
       dataJob.getDeployments() 

Perhaps it is worthwhile do do the same here?

Reply:
Somehow I missed that.